src/schema/posts.ts

```diff
@@ -1729,21 +1729,35 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       const mentions = await getMentions(con, content, userId, sourceId);
 
       const pendingPost: CreateSourcePostModeration = {
-        title,
-        content,
         sourceId,
         type,
         sharedPostId,
         externalLink,
         createdById: userId,
       };
 
-      if (commentary) {
+      const isExternal = !!externalLink;
+
+      if (commentary && type === PostType.Share) {
         await validateCommentary(commentary);
-        pendingPost.titleHtml = generateTitleHtml(commentary, mentions);
+        const commentaryHtml = generateTitleHtml(commentary, mentions);
+
+        if (isExternal) {
+          pendingPost.title = title;
+          pendingPost.content = commentary;
+          pendingPost.contentHtml = commentaryHtml;
+        } else {
+          pendingPost.title = commentary;
+          pendingPost.titleHtml = commentaryHtml;
+        }
       }
```

Comment:
I find it hard to follow with all the if else on what is added and I am sure we can easily forget a field or two in all combinations. What happens if someone sends no commentary and content with type article? Will it go through?

Reply:
I missed during the review that there were three parameters where there should have been only two. Sending a type article is not possible, but on that case, if you send an empty commentary with `share` type for example, a type we accept, it should proceed as we won't care the actual content.

Reply:
> Sending a type article is not possible

You mean from the frontend or in general its prevented somehow on API side?

Reply:
In general, through this process. Type article only comes from post scraper and external links, but then that is created within the system not by sending a `article` type.

Reply:
yes but someone can send it directly to API and what happens then?

Reply:
Rather limit what types can be sent here.

Reply:
Added the checks now. It should not allow types unless defined 🚀